sdk/samples/iot/pnp/pnp_thermostat_component.c

```diff
@@ -273,17 +272,33 @@ bool pnp_thermostat_process_property_update(
     IOT_SAMPLE_LOG("Minimum Temperature: %2f", ref_thermostat_component->minimum_temperature);
     IOT_SAMPLE_LOG("Average Temperature: %2f", ref_thermostat_component->average_temperature);
 
-    // Build reported property message with status.
-    pnp_build_reported_property_with_status(
-        payload,
-        ref_thermostat_component->component_name,
-        property_name->slice,
-        append_double_callback,
-        (void*)&parsed_property_value,
-        (int32_t)AZ_IOT_STATUS_OK,
-        version,
-        twin_response_success,
-        out_payload);
+    az_json_writer jw;
+    az_result rc = az_json_writer_init(&jw, payload, NULL);
+
+    if (az_result_failed(rc = az_json_writer_append_begin_object(&jw))
```

Comment:
nit: Use similar `IOT_SAMPLE_EXIT_IF_AZ_FAILED` pattern here and in `paho_iot_hub_pnp_component_sample.c`, where beneficial.